### ISSUE TITLE
iq/callee: newlines in incoming stanza are parsed as text

### ISSUE THREAD

**Issue by marc0s** [xmppjs/xmpp.js#638]
Hi, this is my first attempt to use `@xmpp/component` so please bear with me if I'm just doing something I shouldn't and this is not a bug :)

I'm trying to create a component that, among other, handles `iq` queries for service discovery, stanzas like

```
<iq xmlns="jabber:client" to="component.localhost" id="x13" type="get" from="alice@localhost/EHN25">
<query xmlns="http://jabber.org/protocol/disco#info" />
</iq>
```

To handle those, I'm testing with
```
xmpp.iqCallee.get('http://jabber.org/protocol/disco#info', 'query', (ctx) => {
	console.log('iqCallee', ctx);
});
```
Is this the expected usage? Because I'm getting a reply with `<error type="modify">`. Searching where the problem might be I found that the incoming stanza is parsed with ` children: [ '\n', [Element], '\n' ],` and that these `children` array does not pass the validation test of `isValidQuery` from `iq/callee.js` because given `child` equals `[child] = stanza.children` (that is, `\n').

If I send the same original XML with `\n` removed, the validation passes (`children` equals just `[[Element]]`) and `return xml(...)` works as expected.

**Comment by sonnyp** [comment]
@marc0s thanks for debugging this, was very helpful

You're doing everything right, it's a bug in xmpp.js :)

https://github.com/xmppjs/xmpp.js/pull/641

**Comment by sonnyp** [comment]
BTW if you're interested, there are some packages here you might find useful: https://github.com/sonnyp/moxc/tree/master/xmpp

I will take some time to move them to xmpp.js

**Comment by sonnyp** [comment]
fixed in [v0.7.1](https://github.com/xmppjs/xmpp.js/releases/tag/v0.7.1)
